### PR TITLE
mtlsChecker: filter out SIMPLE 

### LINKED ISSUE #46555: HTTPS_re-encrypt test flake
https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/46553/integ-pilot-multicluster_istio/1691782056792559616

We create a SIMPLE DR, but the EDS still shows cross-network endpoints: https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/46553/integ-pilot-multicluster_istio/1691782056792559616/artifacts/pilot-989dd9a91575412f97742d3ec/TestTraffic/gateway/HTTPS_re-encrypt/to_b.echo/from_cross-network-primary/_test_context/istio-state31148159/cross-network-primary/istio-ingressgateway-54486f7956-mf6v2_proxy-config.json

Not sure why, at first glance code looks ok. Maybe a caching issue. cc @stevenctl

### PR DESCRIPTION
**Pull request by stevenctl**
fixes #46555

### CODE REVIEW

File: pilot/pkg/xds/endpoints/mtls_checker.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ func (c *mtlsChecker) checkMtlsEnabled(ep *model.IstioEndpoint) bool {
 		switch *drMode {
 		case networkingapi.ClientTLSSettings_DISABLE:
 			return false
+		case networkingapi.ClientTLSSettings_SIMPLE:
```
Comment (howardjohn) [inline pilot/pkg/xds/endpoints/mtls_checker.go:56]: I think anything but mutual is false (per https://github.com/istio/istio/issues/46555#issuecomment-1692256637)

File: tests/integration/pilot/common/routing.go [hunk 2]
```diff
@@ -1790,7 +1790,7 @@ spec:
 		},
 		setupOpts: fqdnHostHeader,
 		opts: echo.CallOptions{
-			Count: 1,
+			Count: 5 * t.Clusters().Len(),
```
Comment (howardjohn) [inline tests/integration/pilot/common/routing.go:1793]: isn't there some CallsForMultiNetwork constant somewhere?

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
lgtm, some small comments

File: tests/integration/pilot/common/routing.go (hunk 2 above)
Comment (stevenctl) [reply tests/integration/pilot/common/routing.go:1793]: I thought so too. I can't find anything but `echo.CallsPerWorkload` which is basically static.

File: pilot/pkg/xds/endpoints/mtls_checker.go (hunk 1 above)
Comment (stevenctl) [reply pilot/pkg/xds/endpoints/mtls_checker.go]: So no MUTUAL, only ISTIO_MUTUAL? Check that it's correct once more before I merge.

File: tests/integration/pilot/common/routing.go (hunk 2 above)
Comment (stevenctl) [reply tests/integration/pilot/common/routing.go]: It's all embedded in the test framework. Just needed to remove `Count`.

File: pilot/pkg/xds/endpoints/mtls_checker.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/xds/endpoints/mtls_checker.go]: yep

**Pull request merged**